src/handlers/label-change.ts

```diff
@@ -0,0 +1,55 @@
+import { isUserAdminOrBillingManager } from "../shared/issue";
+import { Context } from "../types/context";
+import { isLabelEditedEvent } from "../types/typeguards";
+
+export async function watchLabelChange(context: Context) {
+  const logger = context.logger;
+  if (!isLabelEditedEvent(context)) {
+    logger.debug("Not a label event");
+    return;
+  }
+
+  const payload = context.payload;
+  const { label, changes, sender } = payload;
+
+  const previousLabel = changes?.name?.from;
+  if (!previousLabel) {
+    throw logger.error("previous label name is undefined");
+  }
+  const currentLabel = label?.name;
+  const triggerUser = sender.login;
+
+  if (!previousLabel || !currentLabel) {
+    return logger.debug("No label name change.. skipping");
+  }
+
+  // check if user is authorized to make the change
+  const hasAccess = await hasLabelEditPermission(context, currentLabel, triggerUser);
+
+  await context.adapters.supabase.label.saveLabelChange({
+    previousLabel,
+    currentLabel,
+    authorized: hasAccess,
+    repositoryId: payload.repository.id,
+    userId: sender.id,
+  });
+  return logger.debug("label name change saved to db");
+}
+
+async function hasLabelEditPermission(context: Context, label: string, caller: string) {
```

Comment:
current code defines label edit permission as having admin/billing manager and also access in the database, that seems weird. it should probably be just admin/billing manager?

Reply:
I don't recall why we went out of our way to enable it in the database as well. There must have been a reason, maybe surfing through the old code/gitblames/linked pulls would uncover the reasoning. 

I suppose we can cautiously delete if a decision needs to be made right now.